presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/Environment.java

```diff
@@ -171,10 +171,14 @@ public void stop()
                 .with(timeout, retry)
                 .with(executorService);
 
-        ImmutableList.copyOf(containers.values())
-                .stream()
-                .filter(DockerContainer::isRunning)
-                .forEach(container -> executor.run(container::tryStop));
+        for (DockerContainer container : ImmutableList.copyOf(containers.values())) {
+            try {
+                executor.runAsync(container::tryStop).get();
+            }
+            catch (InterruptedException | ExecutionException e) {
```

Comment:
If we get `InterruptedException`, we should set a flag and restore it after stopping all the containers.

Reply:
If we get interrupted here then I think we should just break (rethrow it and set interrupted flag). 

If understand this correctly this is called from main thread, so breaking this thread should stop the process (are there non deamon threads?) and containers will be cleaned by ryuk.
Am I right here?

Reply:
Yes @kokosing. That happens on the main thread and containers will be killed when ryuk will lost connection to the running process

Reply:
So why you are not rethrowing interrupted (as runtime exception)?